x-pack/plugins/enterprise_search/public/applications/app_search/utils/role/index.ts

```diff
@@ -60,7 +61,10 @@ export const getRoleAbilities = (role: Account['role']): Role => {
         (Array.isArray(role.ability[action]) && role.ability[action].includes(subject))
       );
     },
-    // TODO: canHaveScopedEngines fn
+    canHaveScopedEngines: (roleType: RoleTypes): boolean => {
```

Comment:
Can we move this outside of `getRoleAbilities` since the only place it's used doesn't need `myRole`?

Reply:
Perhaps something like this below?

```JavaScript
export const roleHasScopedEngines = (roleType: RoleTypes): boolean => {
  const unscopedRoles = ['dev', 'editor', 'analyst']
  return unscopedRoles.includes(roleType)
}
```

Reply:
I'm 1000% on board with moving this to a separate utility since it doesn't really need `myRole`! Just didn't want to potentially make the migration harder - I'll make the change now that you've confirmed it makes sense ✨

Reply:
I'll open a new PR for this since I want to clean up this `roles` folder some